OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs

```diff
@@ -104,7 +104,7 @@ public void PlayCustomAnimation(Actor self, string name, Action after = null)
 		public void PlayCustomAnimationRepeating(Actor self, string name)
 		{
 			var sequence = NormalizeSequence(self, name);
-			DefaultAnimation.PlayThen(sequence, () => PlayCustomAnimationRepeating(self, sequence));
+			DefaultAnimation.PlayRepeating(NormalizeSequence(self, sequence));
```

Comment:
Looks like you are normalizing the sequence twice.

Reply:
Ugh, good catch.